src/tests/ftest/util/daos_core_base.py

```diff
@@ -82,11 +83,24 @@ def run_subtest(self):
         try:
             process.run(cmd, env=env)
         except process.CmdError as result:
-            if result.result.exit_status is not 0:
+            if result.result.exit_status != 0:
                 # fake a JUnit failure output
-                with open(self.subtest_name +
-                          "_results.xml", "w") as results_xml:
-                    results_xml.write('''<?xml version="1.0" encoding="UTF-8"?>
+                self.create_results_xml(result)
+                self.fail(
+                    "{0} failed with return code={1}.\n".format(
+                        cmd, result.result.exit_status))
+
+    def create_results_xml(self, result):
+        """Create a JUnit result.xml file for the failed command.
+
+        Args:
+            result (CmdResult): result of the failed command.
+        """
+        filename = "".join([self.subtest_name, "_results.xml"])
```

Comment:
Just curious why `join()` here instead of just `+`.

Reply:
I tend to follow this recommendation https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
Interesting recommendation.  Thanks for the pointer.